test/e2e/helpers.go

```diff
@@ -692,15 +692,27 @@ func resolveKubetestRepoListPath(version string, path string) (string, error) {
 // that has an existing capi offer image available. For example, if the version is "stable-1.22", the function will set it to the latest 1.22 version that has a published reference image.
 func resolveKubernetesVersions(config *clusterctl.E2EConfig) {
 	linuxVersions := getVersionsInCommunityGallery(context.TODO(), os.Getenv(AzureLocation), capiCommunityGallery, "capi-ubun2-2404")
-	windowsVersions := getVersionsInCommunityGallery(context.TODO(), os.Getenv(AzureLocation), capiCommunityGallery, "capi-win-2019-containerd")
 	flatcarK8sVersions := getFlatcarK8sVersions(context.TODO(), os.Getenv(AzureLocation), flatcarCAPICommunityGallery)
 
-	// find the intersection of ubuntu and windows versions available, since we need an image for both.
 	var versions semver.Versions
-	for k, v := range linuxVersions {
-		if _, ok := windowsVersions[k]; ok {
+
+	// Check if Windows testing is explicitly disabled via TEST_WINDOWS environment variable
+	testWindows := os.Getenv("TEST_WINDOWS")
+	windowsRequired := testWindows != "false"
+
+	if windowsRequired {
+		windowsVersions := getVersionsInCommunityGallery(context.TODO(), os.Getenv(AzureLocation), capiCommunityGallery, "capi-win-2019-containerd")
```

Comment:
Random pet peeve to keep in mind in the future: burying `context.TODO()` inline like this makes it super easy to forget to update once a function is updated to accept a `ctx context.Context`.

I generally prefer adding a single `ctx := context.TODO()` at the top of a function so then the rest of the function can use `ctx` as if it were already a parameter. Then when the function signature adds a context, that one line can be deleted instead of hunting for and replacing every instance of a context throughout the function.